Distributed Tests failing but CI is green

On latest main 94c94948b79875ba5247b5c986afa088d970a49d, CI flow for distributed shows success https://github.com/Lightning-AI/lightning-thunder/runs/23172744261

But looking at the log, there are a few tests that have failed.

Sample
```
=================================== FAILURES ===================================
_ CompileDDPTest.test_fsdp_grad_parity_with_without_bucketing_executor_nvfuser_bucketing_block_zero2 _
/usr/local/lib/python3.10/dist-packages/torch/testing/_internal/common_distributed.py:533: in wrapper
    self._join_processes(fn)
/usr/local/lib/python3.10/dist-packages/torch/testing/_internal/common_distributed.py:752: in _join_processes
    self._check_return_codes(elapsed_time)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

```

Link to log: https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=196660&view=logs&j=47e66f3c-897a-5428-da11-bf5c7745762e&t=97be8351-284a-5dba-49eb-f9fe7c3ed1a2&l=811

--- a/.azure/gpu-tests.yml
+++ b/.azure/gpu-tests.yml
@@ -81,6 +81,7 @@ jobs:
       displayName: 'Sanity check / details'
 
     - bash: |
+        set -ex
         coverage run --source thunder -m \
           pytest thunder/tests/ \
             -m "not standalone" \
@@ -100,6 +101,7 @@ jobs:
       displayName: 'Testing: regular'
 
     - bash: |
+        set -ex
         # these test need to run in single thread as they occurs with CUDA OOM
         coverage run --source thunder -m \
            pytest \
@@ -125,6 +127,7 @@ jobs:
     #  displayName: 'Testing: standalone'
 
     - bash: |
+        set -ex
         # run all found tests in given past as standalone
         bash scripts/run_standalone_tests.sh "thunder/tests/distributed"
         # compile coverage results

